src/host/alloc.c

```diff
@@ -65,6 +65,24 @@ void heap_trace(struct mm_heap *heap, int size)
 	malloc_info(0, stdout);
 }
 
+int memcpy_s(void *dest, size_t dest_size,
+	     const void *src, size_t src_size)
+{
+	if (!dest || !src)
+		return -EINVAL;
+
+	if ((dest + dest_size >= src && dest + dest_size <= src + src_size) ||
+		(src + src_size >= dest && src + src_size <= dest + dest_size))
```

Comment:
I would split this if into two ifs, again with trace_error for each to help debug.

Reply:
Should be done in separate pr.

Reply:
ok, that's fine if you send as a new PR.

Reply:
Removed that part from commit will create separate commit for traces in memcpy